### PR TITLE
CB-16720: Skip detach of stack and DB if not present instead of throwing errors, allowing for more flexible resize recovery.

### PR DESCRIPTION
**Pull request by sxxgrc**
Jira: https://jira.cloudera.com/browse/CB-16720

Main objective of this is to not fail the updating of a stack or external DB if they are not found, instead choosing to just skip the step while logging the occurrence appropriately.

This is primarily meant to allow for more flexibility from resize recovery, for instance in the following scenario:

- Resize fails at the beginning of DL creation for the resized DL
- Resized DL therefore has no stack or DB or really anything but the SDX Cluster object
- Detach of resized DL, which occurs at the beginning of resize recovery, should be successful despite these missing

### CODE REVIEW

**Comment by horadla23** [comment]
Is it possible that the resized DL is not found in database but the stack/DB is there on the provider side? I am just asking because we need to make sure we are not leaving anything on provider side which can cause additional costs for customer.

**Comment by sxxgrc** [comment]
> Is it possible that the resized DL is not found in database but the stack/DB is there on the provider side? I am just asking because we need to make sure we are not leaving anything on provider side which can cause additional costs for customer.

No I don't think so, that would imply that the resized DL was created and created those other things without generating the DL object itself, which is actually generated _before_ the resize process itself is started.

Now if we want to start thinking about weird scenarios such as the customer themselves deleting random things or deletion during recovery messing up, resulting in not everything being cleaned up, I guess that theoretically is possible. But we have invested into some other tools such as manual SQL scripts (and the retry action should also help) so I think we are fine.

**Review by horadla23 — APPROVED** [review]

**Review by yg-gd — APPROVED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachDetachUtils.java [hunk 1]
```diff
@@ -31,22 +36,32 @@ public void updateClusterNameAndCrn(SdxCluster sdxCluster, String newName, Strin
...
-                        0L, originalName, initiatorUserCrn, newName, newCrn
-                )
+            regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+            () -> {
+                try {
+                    stackV4Endpoint.updateNameAndCrn(
+                            0L, originalName, initiatorUserCrn, newName, newCrn
+                    );
+                } catch (NotFoundException e) {
+                    LOGGER.warn("Stack not found for original name: '" + originalName + "'. Skipping update of stack.");
```
Comment (lacikaaa) [inline datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachDetachUtils.java:46]: could you add `e` to the logger so we won't loose any stacktrace? same for line 62

**Review by lacikaaa — CHANGES_REQUESTED** [review]

File: datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachDetachUtils.java (hunk 1 above)
Comment (sxxgrc) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachDetachUtils.java:46]: Yeah doing that now

Comment (sxxgrc) [reply datalake/src/main/java/com/sequenceiq/datalake/service/sdx/attach/SdxAttachDetachUtils.java]: Done!

**Review by lacikaaa — APPROVED** [review]

**Pull request merged**